Framework/CurveFitting/src/Algorithms/VesuvioCalculateMS.cpp

```diff
@@ -141,10 +142,16 @@ void VesuvioCalculateMS::exec() {
       continue;
     }
 
+    // Initialize random number generator
+    const int seed = getProperty("Seed");
+    CurveFitting::MSVesuvioHelper::RandomVariateGenerator rng(seed + int(i));
```

Comment:
I do not see any problem with re-setting (effective) seed for each `spectrum/hist, getProperty(seed) + spectrum_index`, for each thread, as done here .
However, I wonder if we can set the seed only once for each thread, such as `getProperty(seed) + thread_start_spectrum_index`?
Here, I am assuming each thread picks up a different `spectrum` to start with and, thus, distinct `thread_start_spectrum_index`.

Reply:
The spectra are not picked up in sequence so the seed for a specific spectrum would vary. As a consequence, the random numbers generated for one spectrum would be different and so the results. It would be impossible to use a UnitTest in this case.

Reply:
I guess this is due to the runtime scheduling of the loop! 
I see the associated challenges with testing.